### ISSUE TITLE
Problems processing incorrect LangAlt values

### ISSUE THREAD

**Issue by postscript-dev** [Exiv2/exiv2#1481]
**Describe the bug**

The processing problems fall into three categories.
1. No language value
2. Empty language value
3. Mismatched and/or incorrect positioning of quotation marks

**To Reproduce**

The problems have been observed on the `old-master` and the `0.27-maintenance` branches (though expected on previous branches also).

1. No language value
This causes a segmentation fault when running
```bash
exiv2 -M "set Xmp.dc.title lang= test1-1" any_image.jpeg
exiv2 -M "set Xmp.dc.title lang=\" test1-2" any_image.jpeg
```
This applies to any LangAlt tag and any image.

2. Empty language value
This is found when running
```bash
exiv2 -M "set Xmp.dc.title lang=\"\" test2" any_image.jpeg
```
This applies to any LangAlt tag and any image.

3. Mismatched and/or incorrect positioning of quotation marks
e.g.
```bash
exiv2 -M "set Xmp.dc.title lang=\"\"test3-1" any_image.jpeg

exiv2 -M "set Xmp.dc.title lang=\"test3-2" any_image.jpeg

exiv2 -M "set Xmp.dc.title lang=\"en-UK test3-3" any_image.jpeg
exiv2 -M "set Xmp.dc.title lang=en-US\" test3-4" any_image.jpeg

exiv2 -M "set Xmp.dc.title lang=test3-5\"" any_image.jpeg
exiv2 -M "set Xmp.dc.title lang=test3-6\"\"" any_image.jpeg

```

This applies to any LangAlt tag and any image.

**Expected behavior**

Exiv2 is expected to cope with a LangAlt statement even if it is not well formed - usually by throwing an exception. The examples in 2) and 3) are accepted and returned by the API, giving the impression that they are valid.

**Desktop (please complete the following information):**
 - OS: Windows 10 (MinGW64 on MinGW/MSYS2) - expected on all
 - Compiler & Version: g++.exe (Rev6, Built by MSYS2 project) 10.2.0
 - Compilation mode and/or compiler flags: Standard CMakeLists.txt values

**Additional context**

The problem occurs because of issues in the [src/value.cpp](https://github.com/Exiv2/exiv2/blob/0.27-maintenance/src/value.cpp#L858) file, namely the
```c++
int LangAltValue::read(const std::string& buf);
```
function.

1. No language value
The segmentation fault is due to not checking the result of
```c++
std::string::size_type pos = buf.find_first_of(' ');
```

2. Empty language value
By not checking if the language is empty, this value is added into `LangAltValue`'s `ValueType` and the `xmpsdk` does not filter this out. I don't think this is allowed under the Xmp specification but others may know better. Using the above example,
```bash
$ exiv2 -px any_image.jpeg
### <SNIP>
Xmp.dc.title                                 LangAlt     1  lang="" test2
```

3. Mismatched and/or incorrect positioning of quotation marks
The accepted language value is defined in [IEFT RFC3066](https://www.ietf.org/rfc/rfc3066.txt) and I think that it should only contain A-Z (either case), 0-9 and a '-'. We are accepting values with quotation marks, although they are filtered out when saving. Using the above example, 
```bash
$ exiv2 -px any_image.jpeg
### <SNIP>
Xmp.dc.title                                 LangAlt     2  lang="en-US" test3-4, lang="en-UK" test3-3
```

**Comment by postscript-dev** [comment]
I will look at this more tomorrow.

Also, the templates used when posting an issue are missing. They were stored in the `master` branch and when this was renamed to `old-master`, it looks like the setting has not transferred. 
You can find the templates [here](https://github.com/Exiv2/exiv2/tree/old-master/.github/ISSUE_TEMPLATE).

**Comment by clanmills** [comment]
@postscript-dev  If I grant you write access to our repos, can you deal with this?

Please read this to understand what's going on with master/old-master.   https://github.com/Exiv2/exiv2/issues/1466#issuecomment-787129589

Here's my situation.  I worked 14 hours/day last week and again this week on BMFF support.  https://github.com/Exiv2/exiv2/pull/1475#issuecomment-787050557

After that I have to ship Exiv2 v0.27.4 RC1 on 2021-03-31.  https://github.com/Exiv2/exiv2/issues/1018#issuecomment-757979745

I was 70 in January and the plan was to retire.

**Comment by postscript-dev** [comment]
> @postscript-dev If I grant you write access to our repos, can you deal with this?
>
> Please read this to understand what's going on with master/old-master. #1466 (comment)

Yes, I intended to submit - hopefully today. I posted so that I could reference this in the testing and commit message and didn't intend to burden you with the work.

> Here's my situation. I worked 14 hours/day last week and again this week on BMFF support.#1473
>
> I was 70 in January and the plan was to retire.

Thank you for all your hard work, it is appreciated.

**Comment by clanmills** [comment]
@postscript-dev That's the man, Mr PostScript.  Thanks.  I'll open the repos now.  I know you'll treat it with the respect it deserves.  Welcome aboard.

When you submit the PR, I'll review and approve.  The PR will merge when the CI is green.
